### PR TITLE
Build-MEMCMPackage

### LINKED ISSUE #11: Create Build-MEMCMPackage
Creates an MEMCM package based on the standard inputs from the package definition file

### PR DESCRIPTION
**Pull request by thoen008**
Resolves #11 Created the function to create a CM application.

The function creates an application with deployment types and detection rules.  It still needs some additional functionality added but does create a application that is based on a script installation. 

Tested using UM5 with Mozilla. Will add some additional issues to capture additional features we will need at some point.

I also add some additional values to the JSON files as I build the function. I have added these fields to the example JSON files.

### CODE REVIEW

**Review by gjslavik — COMMENTED** [review]
@thoen008 Does your PackageConfig.json have the updated/added settings from Jeff's commit the other day?

**Comment by thoen008** [comment]
Yes. I made sure and do an update from master before I made my changes.
__________
Matt Thoen
Computer Management Administrator
Computer & Device Support Service | University of Minnesota
T 612 624-8614
E ***@***.***

**Review by gjslavik — APPROVED** [review]
Nice work!

File: PackageConfig.json [hunk 1]
```diff
@@ -6,6 +6,10 @@
     "versionSource": "Filename",
     "sourcePath": "https://download.mozilla.org/?product=firefox-msi-latest-ssl&os=win64&lang=en-US",
     "configNotes": "General comments about this config file",
+    "owner" : "UMN-AtuoPackager",
```
Comment (jbolduan) [inline PackageConfig.json:9]: Spelling correction needed.

File: UMN-AutoPackager/public/Build-MEMCMPackage.ps1 [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+        [Parameter(Mandatory = $true,
+            HelpMessage = "Input the values of the various packagedefinition.json files.")]
+        [psobject[]]$PackageDefinition,
+        [ValidateNotNull()]
+        [System.Management.Automation.PSCredential]
+        [System.Management.Automation.Credential()]
+        $Credential = [System.Management.Automation.PSCredential]::Empty
+    )
+    begin {
+        Write-Information -MessageData "Command = $($myinvocation.mycommand)" -Tags Meta
```
Comment (jbolduan) [inline UMN-AutoPackager/public/Build-MEMCMPackage.ps1:34]: If I understand Write-Information it drops data into the pipeline.  Do we foresee any issues writing this info into the pipeline?  I'm not really sure, probably not but just wanted to check.

**Review by jbolduan — CHANGES_REQUESTED** [review]
I had a couple comments, I trust your function works as expected and I liked the spats to make the code clearer.  Very well done!

File: UMN-AutoPackager/public/Build-MEMCMPackage.ps1 (hunk 2 above)
Comment (thoen008) [reply UMN-AutoPackager/public/Build-MEMCMPackage.ps1:34]: Based on the information in the book Powershell scripting toolmaking, using Write-Information isn’t any different than using Write-Verbose. By default it doesn't produce any output unless you call the -InformationAction Continue. Then it outputs similar to -verbose. It allows tagging and you can output the data to a variable using -InformationVariable nameofvariable.  I thought I would give it a try and used some examples from the book in the begin. Figuring that might be useful for troubleshooting in the future. I'd be fine if we didn't include it.

**Review by jbolduan — APPROVED** [review]
Discussion in Slack and the added commit fixed/alleviated my review concerns.

**Pull request merged**